components/layout/app-shell/index.tsx

```diff
@@ -154,7 +156,9 @@ export const AppShell: ComponentWithChildren = ({ children }) => {
                 />
               </ButtonGroup>
               <ColorModeToggle />
-              <ConnectWallet colorScheme="primary" />
+              {pathname === "/[wallet]" && !address ? null : (
+                <ConnectWallet colorScheme="primary" />
+              )}
```

Comment:
We show the big box when the wallet is not connected, this is to avoid having that connect wallet button in the middle of the screen and in the top bar at the same time